### PR TITLE
Fix clippy

### PR DESCRIPTION
**Pull request by Lighty0410**
There are couple of things i don't know how to handle: 

1) crate/ui_model/src/config/ui_type.rs - line 27. `SessionLobby(Box<SessionLobbyUi>)` instead of `SessionLobby(SessionLobbyUi)`. Since everything compiles (?) i'm not sure if it may break something. 
2) crate/object_model/src/loaded/object.rs:11:40. Don't know how to handle. 

And also splitter.rs was refactored due to cognitive complexity. Please, take a look.

### CODE REVIEW

File: crate/ui_model/src/config/ui_type.rs [hunk 1]
```diff
@@ -24,5 +24,5 @@ pub enum UiType {
     /// Session Lobby UI.
     ///
     /// This is used for both hosts and joiners.
-    SessionLobby(SessionLobbyUi),
+    SessionLobby(Box<SessionLobbyUi>),
```
Comment (azriel91) [inline crate/ui_model/src/config/ui_type.rs:27]: Wow, `clippy` devs are amazing to have the lint for this

**Review by azriel91 — APPROVED** [review]
Thank you very much :bowing_man:! Also much appreciated that you took the time to refactor `ion.rs` and `splitter.rs` -- would have been happy to let them be (since they're from the `ion-shell` repo).

1. `crate/ui_model/src/config/ui_type.rs`:

    That's fine :slightly_smiling_face:, just so happens where it's used doesn't actually mind its type and is happy with `Box<T>` derefing to `T`.

2. `crate/object_model/src/loaded/object.rs:11:40`

    Gonna choose to go with `#[allow(clippy::too_many_arguments)]` -- I haven't managed to make time to remove the `Object` type (asset loading is a huge beast -- and `Object` was one of the leftovers from an earlier design). I may never find time :stuck_out_tongue:, so this is likely the best call.

If you push the `#[allow(..)]` and uncomment the `-D warnings` in `.github/workflows/ci.yml`, hopefully it passes and we can merge :v:

**Comment by Lighty0410** [comment]
Recheck please! `args: -- -D warnings` is being uncommented.

I have added early return statement in splitter.rs. Can you take a look on it ?  There are tons of predicates and it's so easy to make a mistake :P

**Review by azriel91 — APPROVED** [review]
Yay! and once again thank you very much :bowing_man:

**Pull request merged**